sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveWriterContainers.scala

```diff
@@ -230,7 +230,15 @@ private[spark] class SparkHiveDynamicPartitionWriterContainer(
       val path = {
         val outputPath = FileOutputFormat.getOutputPath(conf.value)
```

Comment:
i think we just need to replace FileOutputFormat.getOutputPath with FileOutputFormat.getTaskOutputPath. because FileOutputFormat.getTaskOutputPath will return $outputPath/_temporary/${attemptId}/.